pkg/synthetictests/platformidentification/types.go

```diff
@@ -40,6 +55,82 @@ func CloneJobType(in JobType) JobType {
 	}
 }
 
+func BuildJobVariants(ctx context.Context, clientConfig *rest.Config) JobVariants {
```

Comment:
I think the error handling here could be better.  I'd suggest changing this to return `(JobVariants, error)`.  I'd also suggest not immediately returning on an error, but collect them all and return any errors at the end, so you make a best effort to collect as much of the job variants as possible.

I would guess we shouldn't fail the test run if we can't collect this information, so the caller could just log the error. But I definitely want evidence in the logs if API calls we expect to basically always work start failing

Reply:
I do agree in general but I have noticed that in some places we try to log things and they don't seem to get logged so I wasn't real sure where to log them.. I definitely agree we shouldn't fail.  In this function the only place that returns on error is the NewForConfig, seems unlikely that if configClient.NewForConfig fails the later kubernetes.NewForConfig would succeed.

Reply:
Ah I see, I didn't look closely enough, returning always on the config client erroring seems fine.

You should be able to use the e2e.Logf helpers in origin to log output.  The other thing I'm wondering now is if one of the `oc` commands fail if you should even write out the file at all.  If sippy starts consuming this, we don't want to update a prow job's variants if the data isn't complete.